src/main/java/org/openrewrite/java/testing/mockito/PowerMockitoWhenNewToMockito.java

```diff
@@ -0,0 +1,141 @@
+                J ret = super.visitMethodDeclaration(method, ctx);
+                List<J.FieldAccess> mockArguments = getCursor().getMessage("POWERMOCKITO_WHEN_NEW_REPLACED");
+                if (mockArguments != null && ret instanceof J.MethodDeclaration) {
+                    doAfterVisit(removeMockUsagesVisitor(mockArguments, method.toString()));
+
+                    J.MethodDeclaration retM = (J.MethodDeclaration) ret;
+
+                    // onlyIfReferenced=false as `maybeAddImport` doesn't seem to find the type referred to in a try statement
+                    // see https://github.com/openrewrite/rewrite/issues/5187
+                    maybeAddImport("org.mockito.MockedConstruction", false);
+
+                    for (J.FieldAccess mockArgument: mockArguments) {
+                        String mockedClassName = ((J.Identifier) mockArgument.getTarget()).getSimpleName();
+                        String variableNameForMock = VariableNameUtils.generateVariableName("mock" + mockedClassName, updateCursor(ret), VariableNameUtils.GenerationStrategy.INCREMENT_NUMBER);
+                        JavaTemplate template = JavaTemplate.builder(String.format("try (MockedConstruction<%s> %s = Mockito.mockConstruction(%s.class)) { } ", mockedClassName, variableNameForMock, mockedClassName))
+                                .contextSensitive()
+                                .imports("org.mockito.MockedConstruction")
+                                .javaParser(JavaParser.fromJavaVersion().classpathFromResources(ctx, "mockito-core"))
+                                .build();
+                        J.MethodDeclaration applied = template.apply(getCursor(), method.getCoordinates().replaceBody());
+                        J.Try tryy = (J.Try) applied.getBody().getStatements().get(0);
+                        retM = applied.withBody(applied.getBody().withStatements(Collections.singletonList(tryy.withBody(retM.getBody()))));
+                    }
+                    return autoFormat(retM, ctx);
+                }
+                return ret;
+            }
+
+            private JavaIsoVisitor<ExecutionContext> removeMockUsagesVisitor(List<J.FieldAccess> mockArguments, String inMethodSignature) {
+                Set<String> mockedClassNames = mockArguments.stream().map(fa -> ((J.Identifier) fa.getTarget()).getSimpleName()).collect(Collectors.toSet());
+                return new JavaIsoVisitor<ExecutionContext>() {
+                    @Override
+                    public J.VariableDeclarations visitVariableDeclarations(J.VariableDeclarations multiVariable, ExecutionContext ctx) {
+                        J.VariableDeclarations ret = super.visitVariableDeclarations(multiVariable, ctx);
+                        Cursor containingMethod = getCursor().dropParentUntil(x -> x instanceof J.MethodDeclaration);
+                        if (!inMethodSignature.equals(containingMethod.getValue().toString())) {
```

Comment:
I don't think you have to use a String, since we just fixed UUID "stays" the same. Using the `method` instead of the signature works just fine:

```suggestion
                        if (!inMethod.equals(containingMethod.getValue())) {
```

(with `private JavaIsoVisitor<ExecutionContext> removeMockUsagesVisitor(List<J.FieldAccess> mockArguments, J.MethodDeclaration inMethod) {` of course)

Reply:
3b42aae